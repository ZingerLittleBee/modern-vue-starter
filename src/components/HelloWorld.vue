<script setup lang="ts">
import { ref } from 'vue'

defineProps<{ msg: string }>()
const count = ref(0)
</script>

<template>
    <div class="hw-container">
        <div
            class="mb-10 text-6xl font-black italic text-transparent bg-clip-text bg-gradient-to-r from-green-400 to-blue-500 hover:from-pink-500 hover:to-yellow-500"
        >
            {{ msg }}
        </div>

        <p>
            <span>current count: </span>
            <span class="underline decoration-sky-500">{{ count }}</span>
        </p>
        <button class="btn" @click="() => count++">Click Me</button>
    </div>
</template>

<style scoped>
.hw-container {
    @apply h-screen w-screen flex flex-col justify-center items-center space-y-4;
}

.btn {
    @apply py-2 px-4 rounded text-white bg-violet-500 hover:bg-violet-600 active:bg-violet-700 focus:outline-none focus:ring focus:ring-violet-300;
}
</style>
