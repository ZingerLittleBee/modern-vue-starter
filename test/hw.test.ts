import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import HelloWorld from '../src/components/HelloWorld.vue'

// @vitest-environment jsdom
test('mount hello world component', async () => {
    expect(HelloWorld).toBeTruthy()

    const wrapper = mount(HelloWorld, {
        props: {
            msg: 'test-modern-vue-starter'
        }
    })

    expect(wrapper.text()).toContain('test-modern-vue-starter')
    // https://vitest.dev/guide/snapshot.html
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('2')
})
