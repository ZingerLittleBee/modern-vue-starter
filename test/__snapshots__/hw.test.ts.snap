// Vitest Snapshot v1

exports[`mount hello world component 1`] = `
"<div class=\\"hw-container\\" data-v-469af010=\\"\\">
  <div class=\\"mb-10 text-6xl font-black italic text-transparent bg-clip-text bg-gradient-to-r from-green-400 to-blue-500 hover:from-pink-500 hover:to-yellow-500\\" data-v-469af010=\\"\\">test-modern-vue-starter</div>
  <p data-v-469af010=\\"\\"><span data-v-469af010=\\"\\">current count: </span><span class=\\"underline decoration-sky-500\\" data-v-469af010=\\"\\">0</span></p><button class=\\"btn\\" data-v-469af010=\\"\\">Click Me</button>
</div>"
`;
